src/components/ToDoList.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useContext } from 'react';
+import { DispatchContext, StateContext } from './StateContext';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { FilterType } from '../types/FilterType';
+
+export const ToDoList = () => {
+  const { todos, filter } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const filteredTodos: () => Todo[] = () => {
+    switch (filter) {
+      case FilterType.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case FilterType.COMPLETED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const useFilteredTodos = filteredTodos();
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {/* This is a completed todo */}
+      {useFilteredTodos.map(todo => (
```

Comment:
There is that pattern in react that when you have some kind of list component then thee is always some item component. What i would suggest here is to map through todos and return new component `ToDoItem`, to make the code cleaner